### DISCUSSION TITLE
`StaticResourceExtension` and `ClrPropertyInfo`

### DISCUSSION THREAD

**Discussion by workgroupengineering** [AvaloniaUI/Avalonia#11940] [Q&A]
While debugging my app, I saw this:

![immagine](https://github.com/AvaloniaUI/Avalonia/assets/12531229/24bab9e0-bca4-4c48-804e-82da4600ca8e)

When targetProperty is `ClrPropertyInfo` targetType is null.
Is this an expected behavior?

**Comment by maxkatz6** [comment]
Under what conditions does it happen?

**Reply by workgroupengineering** [reply]
```xaml
   <TextBox Text="{Binding MyProp, Converter={StaticResource cv}}"/>
```

The target  is Converter property.

**Reply by workgroupengineering** [reply]
@maxkatz6 also in this case

```xaml
    <local:TestDataContext StringProperty="Hello"
                           DecimalValue="{StaticResource Value}">
      <local:TestDataContext.ObservableCollectionProperty>
        <sys:String>a</sys:String>
        <sys:String>b</sys:String>
        <sys:String>c</sys:String>
        <sys:String>e</sys:String>
      </local:TestDataContext.ObservableCollectionProperty>
    </local:TestDataContext>
```
I think there was a typo in this line

https://github.com/AvaloniaUI/Avalonia/blob/92b6a4c99ffab361c58b87c248d98e133f72d6a1/src/Markup/Avalonia.Markup.Xaml/MarkupExtensions/StaticResourceExtension.cs#L42

PropertyType was used instead of IPropertyType.
What do you think? Can I make a PR?

**Reply by maxkatz6** [reply]
Looks like it should be a pair of IPropertyType (Avalonia interface) and PropertyInfo (BCL type) checks.
I am not too familiar with this part, should be well tested.

**Reply by workgroupengineering** [reply]
Starting with this test:

https://github.com/AvaloniaUI/Avalonia/blob/a182e1fd98edf858970a9efaf5dc24ffa950a41b/tests/Avalonia.Markup.Xaml.UnitTests/Xaml/XamlIlTests.cs#L232-L241

I've seen that `XamlX` always returns `ClrPropertyInfo` at runtime.

@kekekeks can you confirm it?

if true, my deduction would be correct.
